### PR TITLE
Internal: Reactively update checklist status after changes [ED-15827]

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 1]
```diff
@@ -358,6 +358,35 @@ test.describe( 'Launchpad checklist tests', () => {
...
+		await page.locator( '.button.media-button' ).click();
+		await page.locator( '.elementor-panel button', { hasText: 'Save Changes' } ).click();
+
+		await page.locator( checklistHelper.getStepItemSelector( 'add_logo', '[data-is-checked="true"]' ) ).waitFor();
+
+		await page.locator( '.elementor-control-site_logo .elementor-control-media__content' ).hover();
+		await page.locator( '.elementor-control-site_logo .elementor-control-media__content .elementor-control-media__remove' ).click();
+		await page.locator( '.elementor-panel button', { hasText: 'Save Changes' } ).click();
+
+		await page.locator( checklistHelper.getStepItemSelector( 'add_logo', '[data-is-checked="false"]' ) ).waitFor();
```
Comment (davseve) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:387]: Please add an assertion

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 2]
```diff
@@ -358,6 +358,35 @@ test.describe( 'Launchpad checklist tests', () => {
...
+		await expect( page.locator( checklistHelper.getStepItemSelector( 'add_logo', '[data-is-checked="false"]' ) ) ).toBeVisible();
+		await checklistHelper.clickStepCta( 'add_logo', 'editor' );
+
+		await page.locator( '.elementor-control-site_logo' ).waitFor();
+		await page.locator( '.elementor-control-site_logo .eicon-plus-circle' ).click();
+		await page.getByRole( 'tab', { name: 'Media Library' } ).click();
+		await page.locator( '.thumbnail' ).first().waitFor();
+		await page.locator( '.thumbnail' ).first().click();
+		await page.locator( '.button.media-button' ).click();
+		await page.locator( '.elementor-panel button', { hasText: 'Save Changes' } ).click();
```
Comment (davseve) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:379]: Place in a helper function

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 3]
```diff
@@ -358,6 +358,35 @@ test.describe( 'Launchpad checklist tests', () => {
...
+		await wpAdmin.openNewPage();
+
+		const checklistHelper = new ChecklistHelper( page, testInfo, apiRequests );
+
+		await checklistHelper.toggleChecklistItem( 'add_logo', 'editor', true );
+		await expect( page.locator( checklistHelper.getStepItemSelector( 'add_logo', '[data-is-checked="false"]' ) ) ).toBeVisible();
+		await checklistHelper.clickStepCta( 'add_logo', 'editor' );
+
+		await page.locator( '.elementor-control-site_logo' ).waitFor();
+		await page.locator( '.elementor-control-site_logo .eicon-plus-circle' ).click();
```
Comment (davseve) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:374]:
```suggestion
		const siteLogo =  page.locator( '.elementor-control-site_logo' );
		await siteLogo.waitFor();
		await siteLogo.locator( '.eicon-plus-circle' ).click();
```

File: tests/playwright/sanity/modules/checklist/checklist.test.ts (hunk 2 above)
Comment (ronenelementor) [reply tests/playwright/sanity/modules/checklist/checklist.test.ts:379]: Done

File: tests/playwright/sanity/modules/checklist/checklist.test.ts (hunk 3 above)
Comment (ronenelementor) [reply tests/playwright/sanity/modules/checklist/checklist.test.ts:374]: Done

**Review by davseve — APPROVED** [review]

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**